File: solutions/kotlin/word-count/1/src/main/kotlin/WordCount.kt
object WordCount {

    fun phrase(phrase: String): Map<String, Int> {
        return phrase.replace(Regex("[^\\w\\d ']"), " ")
            .lowercase()
            .split(" ")
            .filter { it != "" }
            .map { it.trim(' ', '\'') }
            .groupBy { it }
            .mapValues { it.value.size }

    }
}

File: solutions/go/word-count/1/word_count.go
package wordcount

import "strings"

type Frequency map[string]int

func WordCount(phrase string) Frequency {
	freq := Frequency{}
	separators := "&@$%^&:!?,.; \t\n"
	for _, word := range strings.FieldsFunc(phrase, func(r rune) bool {
		return strings.ContainsRune(separators, r)
	}) {
		if word != "" && word != "'" {
			word = strings.ToLower(strings.Trim(word, "'"))
			freq[word]++
		}
	}
	return freq

}
